test/e2e/wireguard_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"antrea.io/antrea/pkg/agent/config"
+)
+
+// TestWireGuardTunnelConnectivity checks that Pod traffic across two Nodes over
+// the WireGuard tunnel, by creating multiple Pods across distinct Nodes and having
+// them ping each other.
+func TestWireGuardTunnelConnectivity(t *testing.T) {
+	skipIfProviderIs(t, "kind", "pkt_mark does not work with ovs netdev datapath")
```

Comment:
Does the limitation still exist?

Reply:
I think so. And I don't think they plan to change that either as it would be a non-trivial change. Am I missing something? @williamtu

Reply:
I know the limitation on OVS side is still there. I mean the implemenation doesn't require policy routing now, so it should no longer rely on pkt_mark.

Reply:
With WireGuard running in userspace we can now run the tests in kind clusters.

Reply:
yes, pkt_mark only works for OVS kernel data path. OVS netdev or, userspace datapath, doesn't support pkt_mark.

Reply:
@williamtu thanks for confirming